src/osvr/USBSerial/USBSerialDevInfo_MacOSX.h

```diff
@@ -51,51 +54,87 @@ namespace usbserial {
     namespace {
 
         /**
-         * Given the name of a device, this function will create a
-         *USBSerialDevice
-         * object. If the device can be created successfully it will return
-         * @c boost::none.
-         *
-         * @param device The name of the device to create (e.g., "ttyACM0")
-         *
-         * @return an optional USBSerialDevice
+         * Helper function that returns an IOKit iterator for all serial ports on the system.
+         * @param matchingServices pointer which will be set to the value of the iterator
+         * @return 0 on success, -1 on failure
          */
-        boost::optional<USBSerialDevice>
-        make_USBSerialDevice(const std::string &device) {
-            //FIXME: STUB
-            return boost::none;
+        int findSerialPorts(io_iterator_t *matchingServices) {
+
+            kern_return_t kernResult;
+            CFMutableDictionaryRef classesToMatch;
+            // Query IOKit for services matching kIOSerialBSDServiceValue
+            classesToMatch = IOServiceMatching(kIOSerialBSDServiceValue);
```

Comment:
Combine the declaration on line 64 with the definition on line 66: `CFMutableDictionaryRef classesToMatch = IOServiceMatching(kIOSerialBSDServiceValue);`

In general, try to limit the scope of variables as much as possible.